pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -106,141 +173,263 @@ func (c *Compactor) stopping(_ error) error {
 	return nil
 }
 
-// TODO Get fpRange owned by the compactor instance
-func NoopGetFingerprintRange() (uint64, uint64) { return 0, 0 }
+type Series struct { // TODO this can be replaced with Job struct based on Salva's ring work.
+	tableName, tenant string
+	labels            labels.Labels
+	fingerPrint       model.Fingerprint
+	chunks            []chunk.Chunk
+	from, through     model.Time
+	indexPath         string
+}
 
-// TODO List Users from TSDB and add logic to owned user via ring
-func NoopGetUserID() string { return "" }
+func makeChunkRefs(chksMetas []tsdbindex.ChunkMeta, tenant string, fp model.Fingerprint) []chunk.Chunk {
+	chunkRefs := make([]chunk.Chunk, 0, len(chksMetas))
+	for _, chk := range chksMetas {
+		chunkRefs = append(chunkRefs, chunk.Chunk{
+			ChunkRef: logproto.ChunkRef{
+				Fingerprint: uint64(fp),
+				UserID:      tenant,
+				From:        chk.From(),
+				Through:     chk.Through(),
+				Checksum:    chk.Checksum,
+			},
+		})
+	}
 
-// TODO get series from objectClient (TSDB) instead of params
-func NoopGetSeries() *v1.Series { return nil }
+	return chunkRefs
+}
 
-// TODO Then get chunk data from series
-func NoopGetChunks() []byte { return nil }
+// TODO Revisit this step once v1/bloom lib updated to combine blooms in the same series
+func buildBloomBlock(bloomForChks v1.SeriesWithBloom, series Series) (bloomshipper.Block, error) {
+	localDst := createLocalFileName(series)
 
-// part1: Create a compact method that assumes no block/meta files exists (eg first compaction)
-// part2: Write logic to check first for existing block/meta files and does above.
-func (c *Compactor) compactNewChunks(ctx context.Context, dst string) (err error) {
-	// part1
-	series := NoopGetSeries()
-	data := NoopGetChunks()
+	//write bloom to a local dir
+	builder, err := v1.NewBlockBuilder(v1.NewBlockOptions(), v1.NewDirectoryBlockWriter(localDst))
+	if err != nil {
+		level.Info(util_log.Logger).Log("creating builder", err)
+		return bloomshipper.Block{}, err
+	}
 
-	bloom := v1.Bloom{ScalableBloomFilter: *filter.NewDefaultScalableBloomFilter(0.01)}
-	// create bloom filters from that.
-	bloom.Add([]byte(fmt.Sprint(data)))
+	err = builder.BuildFrom(v1.NewSliceIter([]v1.SeriesWithBloom{bloomForChks}))
+	if err != nil {
+		level.Info(util_log.Logger).Log("writing bloom", err)
+		return bloomshipper.Block{}, err
+	}
 
-	// block and seriesList
-	seriesList := []v1.SeriesWithBloom{
-		{
-			Series: series,
-			Bloom:  &bloom,
-		},
+	blockFile, err := os.Open(filepath.Join(localDst, "bloom"))
```

Comment:
nit: let's make constants public and use them from package `v1` instead of using string values "bloom"
```
const (
	bloomFileName  = "bloom"
	seriesFileName = "series"
)
```